api-platform/core
# Support HTTP "From" header when using Gedmo Blameable extension

| Q             | A
| ------------- | ---
| Bug fix?      | no 
| New feature?  | yes 
| BC breaks?    | no
| Deprecations? | no
| Tickets       | fixes #3383 
| License       | MIT
| Doc PR        | api-platform/docs#...

**Description**
There are situations when a user making a request does not necessarily need to be authenticated with an API. We use this functionality in our internal APIs to track the users that have made CRUD requests. I think this functionality should be upstream rather than in a separate github extension repository or maintained locally within a private repository.

**Example**
Similar to the FOSUserBundle integration, this integration would be controlled by a flag for another Bridge. By default, it would accept this "From" header on any POST or PATCH request. This would be added to the Swagger documentation for these endpoints automatically.

**Testing Notes**
1. Add the following to the api-platform.yaml config
`doctrine_gedmo_extensions:
        from_header_enabled: true
        from_header_unmatch: ['/_action/']`
2. Make sure the gedmo blameable extension is installed 
3. Add the blameable annotation to a field in an entity
4. Submit an update with a "From" header
5. See the "From" user being written to the entity

**Unit tests will be added to the PR after I get a thumbs up from an API Platform developer**
